src/main/proto/ess.proto

```diff
@@ -0,0 +1,64 @@
+syntax = "proto3";
+option java_multiple_files = true;
+package org.ess;
+
+service EncryptedStorageService {
+  rpc get(GetRequest) returns (GetResponse);
+  rpc put(PutRequest) returns (PutResponse);
+  //TODO: Add getKeysSummary (Gives version numbers for all keys in a store) and list by prefix API's
+}
+
+message GetRequest {
+  /*
+    All API's operate within a single storeId.
+    StoreId is keyspace identifier. Ref: https://en.wikipedia.org/wiki/Keyspace_(distributed_data_store)
+    It is up to clients to use single or multiple store's for their usage.
+   */
+  string storeId = 1;
+  string key = 2;
+}
+message GetResponse {
+  bool success = 1;
+  KeyValue value = 2;
+
+  // Only present when success = false
+  optional ErrorResponse errorResponse = 3;
+}
+
+message PutRequest {
+  string storeId = 1;
+  /*
+  If present, write will only succeed if current DB globalVersion against a storeId is
+  same as in request.
+   */
+  optional int64 globalVersion = 2;
```

Comment:
Could you flesh out all the docs in this file? At a high-level, the versioning scheme should be defined.

Reply:
Sounds good, i will write a basic explanation in docs.
Hoping to write a much more detailed api doc after impl though.